map/place_page_info.cpp

```diff
@@ -261,7 +262,24 @@ kml::LocalizableString Info::FormatNewBookmarkName() const
   }
   else if (!m_uiTitle.empty())
   {
-    kml::SetDefaultStr(bookmarkName, m_uiTitle);
+    if (IsMyPosition())
+    {
+      //TODO: locale-specific datetime formatting
+      // %X is HH:MM:SS and optional am/pm; %x is locale-specific date e.g. 18.01.2022 or 01/18/2022
+      //std::string dateTime = base::FormatDateTime("%X %x", std::time(nullptr));
+      //dateTime.replace(5, 3, ""); // remove seconds
+
+      const std::string dateTime = base::FormatDateTime("%H:%M %d.%m.%y", std::time(nullptr));
+      //TODO: add variadic args to GetLocalizedString and fill in placeholders on ios/android side
+      std::string name = platform::GetLocalizedString("core_my_position_bookmark");
+      strings::ReplaceAll(name, "%@", "%s"); // %@ -> %s
+      strings::ReplaceAll(name, "$@", "%s"); // %2$@ -> %2$s
+      name = boost::str(boost::format(name) % dateTime);
```

Comment:
I thought so initially, but it'll be kind of hacky solution tailored to this single use case only. It would be better to have a proper universal placeholder formatting, supporting multiple args and various placeholders formats. Unfortunately placeholder syntax is different in android and ios, hence an ugly way to normalize them..
I will see if it'll be hard or not to pass the replacement args back to ios/android and do the formatting there.

Reply:
No, we don't need a fixed format here. Let's stick to user's locale/device settings.

Reply:
Yeap I think I'll add e.g. `platform::GetLocalizedMyPositionBookmarkName()` which will do all DT formatting and placeholder substitution on platforms' side.